project/src/com/google/daggerqueryui/scripts/binding_graph.js

```diff
@@ -47,22 +47,92 @@ const bindingGraph = (function() {
    */
   let needsToRedraw = true;
 
+  /**
+   * An enum that defines which nodes from the edge will be highlighted.
+   * @type {Readonly<{highlightTarget: number, noHighlight: number, highlightSource: number}>}
+   */
+  const EdgeType = Object.freeze({'highlightSource': 1, 'highlightTarget': 2, 'noHighlight': 3});
+
+  /**
+   * Contains information that can be used by the edges constructor to change their style.
+   * @param {EdgeType} type
+   * @param {string} baseColor
+   * @param {string} highlightColor
+   */
+  function EdgeStyle(type, baseColor, highlightColor) {
+    this.type = type;
+    this.baseColor = baseColor;
+    this.highlightColor = highlightColor;
+  }
+
+  /**
+   * Contains information that can be used by the nodes constructor to change their style.
+   * @param {string} colour a hex string representing the background color of the node
+   * @param {string} shape the shape defines what the node looks like
+   */
+  function NodeStyle(colour, shape) {
+    this.colour = colour;
+    this.shape = shape;
+  }
+
+  const GRAY_COLOR = '#c8c8c8';
+  const ELLIPSE_SHAPE = 'ellipse';
+  const queueWithColours = ['#92cfff', '#fbaed2', '#93d145', '#f8cc43'];
+
+  /**
+   * Retrieves the first colour from the queue and places it at the end of the queue.
+   * @return {string} the first colour from the queue
+   */
+  function extractColor() {
+    const pickedColor = queueWithColours[0];
+    queueWithColours.push(pickedColor);
+    queueWithColours.shift();
```

Comment:
Since `.shift()` returns the first element, you can save one line :stuck_out_tongue: 
```js
const pickedColor = queueWithColours.shift();
queueWithColours.push(pickedColor);
```

Reply:
🚀 Thanks